src/components/UserAccess/UserAccessForm/RoleSection.tsx

```diff
@@ -2,28 +2,30 @@ import React from 'react';
 import { ExpandableSection, FormSection } from '@patternfly/react-core';
 import { useField } from 'formik';
 import HelpPopover from '../../../components/HelpPopover';
+import { useRoleMap } from '../../../hooks/useRole';
 import DropdownField from '../../../shared/components/formik-fields/DropdownField';
-import { WorkspaceRole } from '../../../types';
+import { NamespaceRole } from '../../../types';
 import { PermissionsTable } from './PermissionsTable';
-
 import './RoleSection.scss';
 
-const dropdownItems = [
-  { key: 'contributor', value: 'contributor' },
-  { key: 'maintainer', value: 'maintainer' },
-  { key: 'admin', value: 'admin' },
-];
-
 export const RoleSection: React.FC<React.PropsWithChildren<unknown>> = () => {
-  const [{ value: role }] = useField<WorkspaceRole>('role');
+  const [{ value: role }] = useField<NamespaceRole>('role');
+  const [roleMap, roleMapLoading] = useRoleMap();
+
+  const dropdownItems = [
+    ...Object.entries(roleMap).map(([key, value]) => ({
+      key,
+      value,
+    })),
+  ];
 
   return (
     <>
       <FormSection title="Assign role">
         <DropdownField
           className="role-section"
           name="role"
-          placeholder="Select role"
+          placeholder={roleMapLoading ? 'Loading...' : 'Select role'}
```

Comment:
As suggested, we remove the spinner disable the drop-down with 'Loading' before the roleMap is ready.